src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -544,7 +549,20 @@ public void runRemotely(Step s) throws Throwable {
         conn.setRequestProperty("Content-Type", "application/octet-stream");
         conn.setDoOutput(true);
         Init2.writeSer(conn.getOutputStream(), Arrays.asList(token, s));
-        Throwable error = (Throwable) Init2.readSer(conn.getInputStream(), null);
+        Throwable error;
+        try {
+            error = (Throwable) Init2.readSer(conn.getInputStream(), null);
+        } catch (IOException e) {
+            if (conn.getErrorStream() != null) {
+                try {
+                    String errorMessage = IOUtils.toString(conn.getErrorStream(), StandardCharsets.UTF_8);
+                    e.addSuppressed(new IOException("Response body: " + errorMessage));
+                } catch (IOException e2) {
+                    e.addSuppressed(e2);
+                }
+            }
+            error = e;
```

Comment:
or just
```suggestion
            throw e;
```

Reply:
Done in [869ed42](https://github.com/jenkinsci/jenkins-test-harness/pull/422/commits/869ed42113bfc9b6f3d3a4f70e26fa1841e7fc3e).